src/main/java/org/apache/datasketches/memory/BaseBuffer.java

```diff
@@ -143,11 +92,7 @@ public final BaseBuffer setPosition(final long position) {
    * @param position the given current position.
    * @return BaseBuffer
    */
-  public final BaseBuffer setAndCheckPosition(final long position) {
-    checkInvariants(start, position, end, capacity);
-    pos = position;
-    return this;
-  }
+  BaseBuffer setAndCheckPosition(long position);
```

Comment:
Do these variants of the interface that check arguments imply that the client typically would expect and handle a possible `IllegalArgumentException`?  How does this differ in severity from an `AssertionError`?

Reply:
There are two types of checks: "check(blah)" and "assert(blah)".   Asserts only apply when running test or if the JVM flag -ea is set.  The check methods will throw an exception.  Note that checks are primarily used when copying arrays or blocks. but when just setting primitives, I use asserts.  The primary reason is speed during runtime, otherwise, the cost of the check would overwhelm the actual operation.  If the user is concerned about memory violations, all they have to do is set -ea.

Reply:
Thanks for the detailed answer - this makes more sense to me now.